Startup spotlight: Golioth (#6355)
PostHog/posthog.com: contents/blog/startup-golioth.md

Reviewer: Don't have edit capabilities so: Line 16 change "Goliath" to "Golioth"

---
title: "Startup Spotlight: How Golioth builds IoT software"
date: 2023-07-26
author: ["joe-martin"]
showTitle: true
rootpage: /blog
sidebar: Blog
hideAnchor: true
featuredImage: ../images/blog/posthog-engineering-blog.png
featuredImageType: full
category: Startups
---

> Each month, we speak to a team enrolled in [PostHog for Startups](/startups) about the way they work and the challenges they face.

> [Golioth](https://golioth.io/) is a seed stage developer platform which enables hardware companies to build IoT products more easily and quickly. Golioth accomplishes this with a cloud platform which can receive sensor data, issue OTA firmware updates and integrate with connected devices — but faces the unique challenge of creating software for new, rapidly evolving hardware projects. 

Author: ```suggestion
[Golioth](https://golioth.io/) is an IoT developer platform that enables engineers to build IoT products and infrastructure quickly and easily. Golioth ensures that developers can securely capture and route sensor data, issue OTA firmware updates, and manage fleets of connected devices — but faces the unique challenge of creating cloud software for new, rapidly evolving hardware projects.
```

“Hardware and software development are very different worlds,” says Golioth Founder and CEO Jonathan Beri. “Hardware? It’s slow. It’s hard. It’s incredibly complicated. And software too has its own unique challenges, even before you add IoT into the mix.”

Golioth has adopted a nuanced approach to product development in order to combine these disciplines effectively — essentially dividing the way teams work based on whether it’s on the firmware & SDK side of the business, or in the cloud product.

“On the firmware side and for our SDKs we have a longer release cycle because those updates get baked into our users’ operating systems,” says Golioth Product Manager Dylan Swartz. “If we ship features every every single day, our customers couldn’t keep up. But on the cloud side we’re much more agile. We run traditional two-week sprints, regularly adjust priorities and release regularly.”

> Regardless of the product however, there are four essential principles which Golioth applies universally, across all aspects of product and feature development…

Reviewer 2: Can we change up for the format of this intro a bit? It needs a bit more pace / urgency – I'm thinking something like [The Guardian's Pass Notes](https://www.theguardian.com/lifeandstyle/2023/jul/19/weekend-warriors-exercise-cardiovascular) minus the snark, if that makes sense?

Author: I'll give it a further edit, but as discussed async that's not really the style I'm going for with this one - nor do I think this content is well suited to it. 

Not ruling it out for a future format switch -- it would be good to try it -- but not this time.

## 2. Hire people who understand the challenges you want to solve

Of course, there’s little point collecting feedback if you don’t have the skills and empathy that are needed to understand and act on it. This, says Jonathan, is why Golioth has focused on hiring engineers who have a keen understanding of _both_ hardware and software development. 

“Imagine a CDN company,” explains Jonathan. “They don’t build large-scale web apps themselves, but they have to understand the challenges that their customers face — and so they build internal expertise by hiring great people. We do the exact same thing.”

Jonathan points to Dylan as a clear example of this in practice. Prior to joining as Golioth’s Product Manager, Dylan had begun his career in software engineering, before moving the building customizable smartphones and tablets for businesses. As a result, he’s able to empathize closely with Golioth’s users and shape the roadmap accordingly.

## 3. Use open-source to bootstrap new products

> Contributing to open source communities is ingrained in Golioth’s culture from an engineering standpoint — but it’s also a competitive advantage which they’ve seized from the start. 

Reviewer 2: ```suggestion
Contributing to open-source communities is ingrained in Golioth’s culture from an engineering standpoint, but it’s also a competitive advantage which they’ve seized from the start. 
```

> “Software on hardware has historically been closed source,” explains Jonathan. “Over the last decade, that’s started to change however and it’s been very exciting. Golioth as a company probably couldn’t have gotten started if it wasn’t for open source projects we could build on.”

Reviewer 2: ```suggestion
“Software on hardware has historically been closed source,” explains Jonathan. “Over the last decade, that’s started to change however and it’s been very exciting. Golioth as a company probably couldn’t have gotten started if it wasn’t for open-source projects we could build on.”
```

> By adopting open source as the basis for much of its work, Golioth (like PostHog) unlocks a lot of advantages that aren’t available to closed source projects. One of these is an opportunity to incorporate features from other, dependent open source projects. 

Reviewer 2: ```suggestion
By adopting open source as the basis for much of its work, Golioth (like PostHog) unlocks a lot of advantages that aren’t available to closed-source projects. One of these is an opportunity to incorporate features from other, dependent open source projects. 
```

---
title: "Startup Spotlight: How Golioth builds IoT software"
date: 2023-07-26
author: ["joe-martin"]
showTitle: true
rootpage: /blog
sidebar: Blog
hideAnchor: true
featuredImage: ../images/blog/posthog-engineering-blog.png
featuredImageType: full
category: Startups
---

> Each month, we speak to a team enrolled in [PostHog for Startups](/startups) about the way they work and the challenges they face.

> [Golioth](https://golioth.io/) is a seed stage developer platform which enables hardware companies to build IoT products more easily and quickly. Goliath accomplishes this with a cloud platform which can receive sensor data, issue OTA firmware updates and integrate with connected devices — but faces the unique challenge of creating software for new, rapidly evolving hardware projects. 

Reviewer 3: ```suggestion
[Golioth](https://golioth.io/) is a seed stage developer platform which enables hardware companies to build IoT products more easily and quickly. Golioth accomplishes this with a cloud platform which can receive sensor data, issue OTA firmware updates and integrate with connected devices — but faces the unique challenge of creating software for new, rapidly evolving hardware projects. 
```

> [Golioth](https://golioth.io/) is a seed stage developer platform which enables hardware companies to build IoT products more easily and quickly. Goliath accomplishes this with a cloud platform which can receive sensor data, issue OTA firmware updates and integrate with connected devices — but faces the unique challenge of creating software for new, rapidly evolving hardware projects. 

Reviewer: Line 16 change "Goliath" to "Golioth"

> [Golioth](https://golioth.io/) is a seed stage developer platform which enables hardware companies to build IoT products more easily and quickly. Goliath accomplishes this with a cloud platform which can receive sensor data, issue OTA firmware updates and integrate with connected devices — but faces the unique challenge of creating software for new, rapidly evolving hardware projects. 

Reviewer: ```suggestion
[Golioth](https://golioth.io/) is an IoT developer platform that enables engineers to build IoT products and infrastructure quickly and easily. Golioth ensures that developers can securely capture and route sensor data, issue OTA firmware updates, and manage fleets of connected devices — but faces the unique challenge of creating cloud software for new, rapidly evolving hardware projects.
```

“On the firmware side and for our SDKs we have a longer release cycle because those updates get baked into our users’ operating systems,” says Golioth Product Manager Dylan Swartz. “If we ship features every every single day, our customers couldn’t keep up. But on the cloud side we’re much more agile. We run traditional two-week sprints, regularly adjust priorities and release regularly.”

Regardless of the product however, there are four essential principles which Golioth applies universally, across all aspects of product and feature development…

## 1. Put customers at the heart of product development

Golioth is an incredibly customer-driven team, and invests a lot of time and effort in speaking with users, collecting feedback and understanding the user experience. 

“We obviously have a long-term vision, but customer feedback always has the highest priority over everything else,” says Dylan. “We even use PostHog to help with this by using feature flags and session replays to test new updates with users who ask, or to debug issues they face.”

> One of the ways this customer focus manifests is in Golioth’s reference designs for hardware projects, which the team creates in order to help hardware engineers. 

Reviewer 3: If interested in linking, this is our reference design showcase :)
```suggestion
One of the ways this customer focus manifests is in Golioth’s [reference designs](https://projects.golioth.io/) for hardware projects, which the team creates in order to help hardware engineers. 
```

“We don’t make hardware ourselves,” says Jonathan, “but we know these references help users to prove out ideas more quickly. That, in turn, means they’re going to be more successful.”

> 

Author: ```suggestion
```

Jonathan points to Dylan as a clear example of this in practice. Prior to joining as Golioth’s Product Manager, Dylan had begun his career in software engineering, before moving the building customizable smartphones and tablets for businesses. As a result, he’s able to empathize closely with Golioth’s users and shape the roadmap accordingly.

## 3. Use open-source to bootstrap new products

Contributing to open source communities is ingrained in Golioth’s culture from an engineering standpoint — but it’s also a competitive advantage which they’ve seized from the start. 

“Software on hardware has historically been closed source,” explains Jonathan. “Over the last decade, that’s started to change however and it’s been very exciting. Golioth as a company probably couldn’t have gotten started if it wasn’t for open source projects we could build on.”

By adopting open source as the basis for much of its work, Golioth (like PostHog) unlocks a lot of advantages that aren’t available to closed source projects. One of these is an opportunity to incorporate features from other, dependent open source projects. 

> > **Community spotlight:** Jonathan credits the [Zephy](https://www.zephyrproject.org/) as the most crucial open source project to Golioth’s own creation, and the team continues to regularly contribute upstream to it as a result. 

Reviewer 3: ```suggestion
> **Community spotlight:** Jonathan credits the [Zephyr](https://www.zephyrproject.org/) as the most crucial open source project to Golioth’s own creation, and the team continues to regularly contribute upstream to it as a result. 
```
